includes/Core/Modules/Module.php

```diff
@@ -388,18 +389,36 @@ public function is_connected() {
 	 * @return array List of datapoints.
 	 */
 	final public function get_datapoints() {
-		return array_keys( $this->get_datapoint_services() );
+		$keys        = array();
+		$definitions = $this->get_datapoint_definitions();
+
+		foreach ( array_keys( $definitions ) as $key ) {
+			$parts = explode( ':', $key );
+			$name  = end( $parts );
+			if ( ! empty( $name ) ) {
+				$keys[ $name ] = $name;
+			}
+		}
+
+		return array_values( $keys );
```

Comment:
Just a suggestion, but what do you think about doing it this way?
We can keep it your way as both should work, I prefer avoiding temporary variables when possible as long as it doesn't sacrifice readability.
```suggestion
return array_unique(
	array_map(
		function ( $key ) {
			// Prevent potential undefined index by ensuring second element is always set.
			list ( , $datapoint ) = explode( ':', $key, 2 ) + array( null, null );
			return $datapoint;
		},
		array_keys( $this->get_datapoint_definitions() )
	)
);
```

Reply:
Nah... this approach is more cumbersome and is harder to read than just small `foreach` loop.